### ISSUE TITLE
Can I chose my own superpixels?

### ISSUE THREAD

**Issue by RenzeLou** [fuy34/superpixel_fcn#19]
When I change the Variable "downsize" to 32, it becomes quite hard for the model to training, that is, the loss reduces slowly (have already trained 800 epochs).
Therefore, can I chose my own superpixels? Is there any special hyparameter that should be changed accordingly?

**Comment by fuy34** [comment]
Hi, 
May I ask how the training looks like if you train with ```downsize=16```? 
I trained with ```downsize=32 and 8``` previously. It should work fine unless I changed something mistakenly during code cleanup.

**Comment by RenzeLou** [comment]
There are several issues, I will list below:
First of all, argument type:
![image](https://user-images.githubusercontent.com/59914581/121699880-eb6b0f80-cb01-11eb-843c-ba6627ec7631.png)

**Comment by RenzeLou** [comment]
There are several issues, I will list below:
First of all, argument type:
![image](https://user-images.githubusercontent.com/59914581/121699880-eb6b0f80-cb01-11eb-843c-ba6627ec7631.png)
It can de fixed by change the type of variable "downsize" in "main.py":
![Uploading image.png…]()

**Comment by RenzeLou** [comment]
First of all, thank you for your replying!
Anyway, I have trained the model with different downsize successfully. I have noticed that:

- The value  of training loss could be higher when I chose bigger downsize
The picture shown below is the model training with downsize=20:
![image](https://user-images.githubusercontent.com/59914581/121845063-7a9d4080-cd17-11eb-9bc0-cd4509c4e462.png)
Whereas I still remember when I chose downsize=8 or 16, the training loss could be quite tiny after several hundred epochs.
- Image size problem
I also found that some problem would occur if I changed the downsize:
![image](https://user-images.githubusercontent.com/59914581/121845877-b2f14e80-cd18-11eb-9cf9-50325d08b64c.png)

![image](https://user-images.githubusercontent.com/59914581/121845767-82a9b000-cd18-11eb-8199-4e47371bd300.png)
As I show above, the shape of the tensor (image) would be incomparable with the model structure due to the 'np.ceil', thus I change 'H_', 'W_' to 'H', 'W' respectively.

Therefore, I'm also not sure whether there are some mistakes when you operate code cleanup.

Best regards!

**Comment by fuy34** [comment]
I may know what happens.

What is your image size from the dataloader? Based on our current design, the ```img_h```  and ```img_w``` should be dividable by ```downsize```. If you keep the default size (208) and choose 32 as downsize, it will get ```208 / 32= 6.5```, which leads to the ceiling problem.  

I was planning to make this compatible (that is why there is np.ceil), but I did not complete it. The main issue is we divide a image to ```h*w ``` grid cells, where ```h = img_h / downsize```, and ```w = img_w /downsize```.  If img_h or img_w is not dividable by ```downsize```, it will lead to incomplete cells, which is an edge-case that requires special treatment.
